### PR TITLE
Custom SlavesFreeSound

### LINKED ISSUE #601: [New Feature] Custom SlavesFreeSound
### First check

- [X] The suggestion is **NOT** related to functionality of Ares, HAres, Kratos or any other YR engine extension.
- [X] The suggested feature / enhancement is **NOT** implemented in Ares/Phobos yet and the existing possibilites are not viable or good enough to achieve the wanted result.
- [X] I agree to elaborate the details if requested and provide thorough testing if the feature is implemented.
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search and read the issue list to find a similar issue and didn't find it.
- [X] I thought out how the feature / enhancement should work and described it in the most detailed and clear way.

### Description:
Now all kind of slaves use general setting "SlavesFreeSound= SlaveWorkerLiberated", so hope it can be custom for each type of slave.
### Possible Implementation:
```ini
[SOMETECHNO]
CustomSlaveFreeSound=<none>        ; kind of voice
```

### PR DESCRIPTION
**Pull request by Super-StarX**
close #601

### CODE REVIEW

File: src/Ext/TechnoType/Hooks.cpp [hunk 1]
```diff
@@ -311,3 +311,17 @@ DEFINE_HOOK(0x4AE670, DisplayClass_GetToolTip_EnemyUIName, 0x8)
...
 DEFINE_PATCH(0x707CF2, 0x55);
+
+// Issue #601
+// Author : TwinkleStar
+DEFINE_HOOK(0x6B0C2C, SlaveManagerClass_FreeSlaves_SlavesFreeSound, 0x5)
+{
+	GET(TechnoClass*, pSlave, EDI);
+
+	auto pTypeExt = TechnoTypeExt::ExtMap.Find(pSlave->GetTechnoType());
+	auto sound = pTypeExt->SlavesFreeSound.Get(RulesClass::Instance()->SlavesFreeSound);
```
Comment (Thrifinesma) [inline src/Ext/TechnoType/Hooks.cpp:322]:
```suggestion
	int sound = pTypeExt->SlavesFreeSound.Get(RulesClass::Instance()->SlavesFreeSound);
```

**Review by Thrifinesma — COMMENTED** [review]
I didn't see other code style issues, just need to be tested

**Comment by NetsuNegi** [comment]
I had test that and it works.

File: docs/Fixed-or-Improved-Logics.md [hunk 2]
```diff
@@ -400,6 +400,17 @@ ForbidParallelAIQueues.Aircraft=no  ; boolean
 ForbidParallelAIQueues.Building=no  ; boolean
 ```
 
+### Custom slaves free sound
+
+- `SlavesFreeSound` is now dehardcoded from `AudioVisual` and can be set to each slave.
```
Comment (Metadorius) [inline docs/Fixed-or-Improved-Logics.md:405]:
```suggestion
- `SlavesFreeSound` is now dehardcoded from `AudioVisual` and can be set individually for each enslaved unit type.
```

**Review by Metadorius — APPROVED** [review]

**Pull request merged**